internal/cri/server/sandbox_run.go

```diff
@@ -451,13 +451,20 @@ func cniNamespaceOpts(id string, config *runtime.PodSandboxConfig) ([]cni.Namesp
 
 // toCNILabels adds pod metadata into CNI labels.
 func toCNILabels(id string, config *runtime.PodSandboxConfig) map[string]string {
-	return map[string]string{
+	labels := map[string]string{
 		"K8S_POD_NAMESPACE":          config.GetMetadata().GetNamespace(),
 		"K8S_POD_NAME":               config.GetMetadata().GetName(),
 		"K8S_POD_INFRA_CONTAINER_ID": id,
 		"K8S_POD_UID":                config.GetMetadata().GetUid(),
 		"IgnoreUnknown":              "1",
 	}
+
+	// Merge PodSandboxConfig labels
+	for k, v := range config.GetLabels() {
```

Comment:
where are these labels coming from? are this the kubernetes pod labels or the containerd labels associated to the pod , or both?

Reply:
it seems is both?

```
 labels=map[app:kindnet controller-revision-hash:7b547bcb47 io.cri-containerd.kind:sandbox io.kubernetes.pod.name:kindnet-7jq5x io.kubernetes.pod.namespace:kube-system io.kubernetes.pod.uid:e6b283d5-4e02-41a3-ab69-cded28329f74 k8s-app:kindnet pod-template-generation:1 tier:node]

```